src/Mod/PartDesign/App/FeatureHole.cpp

```diff
@@ -1346,14 +1348,43 @@ App::DocumentObjectExecReturn *Hole::execute(void)
             double drillPointAngle = Base::toRadians( ( 180.0 - DrillPointAngle.getValue() ) / 2.0 );
             double x, z;
             gp_Pnt newPoint;
-
-            if ( drillPointAngle <= 0 || drillPointAngle > Base::toRadians( 180.0 ) )
-                return new App::DocumentObjectExecReturn("Hole: Invalid drill point angle.");
-
-            computeIntersection(gp_Pnt( 0, -length, 0 ),
-                                gp_Pnt( cos( drillPointAngle ), -length + sin( drillPointAngle ), 0),
-                                gp_Pnt(radius, 0, 0),
-                                gp_Pnt(radiusBottom, -length, 0), x, z);
+            bool isDrillForDepth = DrillForDepth.getValue();
+
+            // the angle is in any case > 0 and < 90 but nevertheless this safeguard:
+            if ( drillPointAngle <= 0 || drillPointAngle >= Base::toRadians( 180.0 ) )
+                return new App::DocumentObjectExecReturn("Hole error: Invalid drill point angle");
+
+            // if option to take drill point size into account
+            // the next wire point is the intersection of the drill dege and the hole edge
+            if (isDrillForDepth) {
+                computeIntersection(gp_Pnt(0, -length, 0),
+                    gp_Pnt(radius, radius * tan(drillPointAngle) - length, 0),
+                    gp_Pnt(radius, 0, 0),
+                    gp_Pnt(radiusBottom, -length, 0), x, z);
+                if (z > 0 || z >= length1)
+                    return new App::DocumentObjectExecReturn("Hole error: Invalid drill point");
+
+                newPoint = toPnt(x * xDir + z * zDir);
+                mkWire.Add(BRepBuilderAPI_MakeEdge(lastPoint, newPoint));
+                lastPoint = newPoint;
+
+                newPoint = toPnt(-length * zDir);
+                mkWire.Add(BRepBuilderAPI_MakeEdge(lastPoint, newPoint));
+                lastPoint = newPoint;
+            }
+            else {
+                x = radiusBottom;
+                z = -length;
+
+                newPoint = toPnt(x * xDir + z * zDir);
+                mkWire.Add(BRepBuilderAPI_MakeEdge(lastPoint, newPoint));
+                lastPoint = newPoint;
+
+                // the end point is the size of the drill tip
+                newPoint = toPnt((-length - radius * tan(drillPointAngle)) * zDir);
+                mkWire.Add(BRepBuilderAPI_MakeEdge(lastPoint, newPoint));
+                lastPoint = newPoint;
+            }
 
             if (z > 0 || z >= length1)

```

Comment:
`length1` is not a very descriptive variable name -- what is it?

Reply:
Yes, this code is hard to read because of these meaningless variables. However, it is not my code. 
Nevertheless I renamed this variable to "lengthCounter" to make clear that it is the length of the counter(sink/bore).